test/test_nestedtensor.py

```diff
@@ -7818,131 +7822,539 @@ def __torch_dispatch__(self, func, types, args=..., kwargs=None):
+COMPILE_BACKWARD_FAILURES = [
+    # Bug: Something is wrongly creating an empty tensor with the jagged layout on the C++ side
+    # for these binary ops
+    XFailRule(
+        error_type=NotImplementedError,
+        error_msg="non-strided meta tensors not supported yet",
+        match_fn=lambda device, dtype, op, sample: (
+            op.full_name
+            in {
+                "__rpow__",
+                "clamp_max",
+                "clamp_min",
+                "float_power",
+                "pow",
+                "sinc",
+            }
+            and "noncontig_holes" not in sample.name
+            and (
+                "(NT, T) broadcasting 1 over ragged" in sample.name
+                or "(NT, T) broadcasting all 1s" in sample.name
+                or "(NT, T) mixed broadcasting" in sample.name
+            )
+        ),
+        name="empty_with_jagged_layout_for_some_binary_ops",
+    ),
+    # Bug: Something is wrongly creating an empty tensor with the jagged layout on the C++ side
+    # for this op when cached seqlen metadata is present
+    XFailRule(
+        error_type=NotImplementedError,
+        error_msg="non-strided meta tensors not supported yet",
+        match_fn=lambda device, dtype, op, sample: (
+            op.full_name
+            in {
+                "special.i1",
+                "special.i1e",
+                "sinc",
+            }
+            and "with_seqlen_cache" in sample.name
+        ),
+        name="empty_with_jagged_layout_with_cached_seqlens",
+    ),
+    # in compile, these complex ops use view_as_real(), which isn't implemented
+    XFailRule(
+        error_type=NotImplementedError,
+        error_msg="aten.view_as_real.default",
+        match_fn=lambda device, dtype, op, sample: (
+            op.full_name in {"cdouble", "cfloat", "chalf"}
+            and "with_seqlen_cache" in sample.name
+        ),
+        name="unimplemented_view_as_real",
+    ),
+    # torch._subclasses.fake_tensor.DataDependentOutputException: aten._local_scalar_dense.default
+    # from item call in clone() -> unbind()
+    XFailRule(
+        error_type=torch._dynamo.exc.Unsupported,
+        error_msg="Backend compiler failed with a fake tensor exception",
+        match_fn=lambda device, dtype, op, sample: (
+            (
+                (
+                    isinstance(op, BinaryUfuncInfo)
+                    and
+                    # don't include unimplemented ops
+                    op.full_name
+                    not in {
+                        "__rsub__",
+                        "complex",
+                        "floor_divide",
+                        "polar",
+                        "rsub",
+                    }
+                )
+                or op.full_name
+                in {
+                    "__rpow__",
+                    "clamp_max",
+                    "clamp_min",
+                    "float_power",
+                    "pow",
+                    "sinc",
```

Comment:
this doesn't look like a binary op :P should we move it out into a separate XFailRule?

Reply:
haha good point - I think I'll just rename the rule to be more accurate